knife/lib/chef/knife/bootstrap.rb

```diff
@@ -712,6 +713,9 @@ def do_connect(conn_options)
           conn_options[:sudo_password] = sudo_password
 
           retry
+        elsif e.message =~ /Sudo failed: \S+/
```

Comment:
I think it would be better to handle this message in [train](https://github.com/inspec/train/blob/62a8b1e0b5f6e2cc88b0ab2e4e9101c571bc99df/lib/train/extras/command_wrapper.rb#L73-L86) 
rather than doing ssh literal text parsing both here and hidden in the train lib - we could supply a `reason` of `:sudo_access_failed`, and handle that here.

Reply:
ok, I will do that same. Thanks

Reply:
@marcparadise I have raised this PR on train repo - https://github.com/inspec/train/pull/690 with required changes. Please take a look

Reply:
Since the changes required have been done as part of this [inspec/train#690](https://github.com/inspec/train/pull/690).. This PR can be closed, as it does not contain any changes